compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -2004,7 +2004,12 @@ object Parsers {
           }
 
           val finalizer =
-            if (in.token == FINALLY) { in.nextToken(); subExpr() }
+            if (in.token == FINALLY) {
+              in.nextToken();
+              val expr = subExpr()
+              if expr.span.exists then expr
```

Comment:
Instead of checking the span, can't we check if the tree is empty/null?

Reply:
Both for `finally ;` and `finally null` the `expr` is `Literal(Constant(null))` but only the latter has a span.